src/AutoSubscriptions.ts

```diff
@@ -245,28 +245,35 @@ function makeAutoSubscribeDecorator(shallow = false, defaultKeyValues: string[])
                 return existingMethod.apply(this, args);
             }
 
-            // Let the handler know about this auto-subscriptions, then proceed to the existing method.
-
-            // Default to Key_All if no @key parameter.
-            let specificKeyValues = defaultKeyValues;
-
-            // Try to find an @key parameter in the target's metadata.
+            // Try to find an @key parameter in the target's metadata and form initial Key(s) from it/them.
+            let keyParamValues: (string | number)[] = [];
             if (metaForMethod.hasIndex) {
-                let keyArg: number | string = args[metaForMethod.index];
-
-                if (_.isNumber(keyArg)) {
-                    keyArg = keyArg.toString();
-                }
-
-                assert(keyArg, `@key parameter must be given a non-empty string or number: ` +
-                    `"${ methodNameString }"@${ metaForMethod.index } was given ${ JSON.stringify(keyArg) }`);
-
-                assert(_.isString(keyArg), `@key parameter must be given a string or number: ` +
-                    `"${ methodNameString }"@${ metaForMethod.index }`);
-
-                specificKeyValues = [keyArg];
+                keyParamValues = metaForMethod.indexes.map(index => {
+                    let keyArg: number | string = args[index];
+
+                    if (_.isNumber(keyArg)) {
+                        keyArg = keyArg.toString();
+                    }
+    
+                    assert(keyArg, `@key parameter must be given a non-empty string or number: ` +
+                        `"${ methodNameString }"@${ index } was given ${ JSON.stringify(keyArg) }`);
+    
+                    assert(_.isString(keyArg), `@key parameter must be given a string or number: ` +
+                        `"${ methodNameString }"@${ index }`);
+                    
+                    return keyArg;
+                });
             }
 
+            // Form a list of keys to trigger.
+            // If we have @key values, put them first, then append the @autosubscribewithkey key to the end.
+            // If there are multiple keys in the @autosubscribewithkey list, go through each one and do the
+            // same thing (@key then value).  If there's neither @key nor @autosubscribewithkey, it's Key_All.
+            const specificKeyValues: string[] = (defaultKeyValues && defaultKeyValues.length > 0) ?
```

Comment:
Feels like the condition should check if there are `keyParamValues` instead, and fallback to `defaultKeyValues`.

Reply:
It's the same ugly ternary either way, unfortunately.  I wrote it out several different ways before giving up.